### DISCUSSION TITLE
Use `.gitignore` patterns by default

### DISCUSSION THREAD

**Discussion by babakfp** [eslint/eslint#17956] [General]
Hi

I want to discuss the idea of automatically not checking certain files when using ESLint. To put it simply, ESLint will skip checking files if they are listed in the `.gitignore` file.

There is no need to lint patterns found in `.gitignore` at all. So, ignoring them by default can clean up the `.eslintignore` file from a lot of projects and it can remove the need and maintenance of syncing the content in both `.gitignore` and `.eslintignore` files.

[Prettier](https://prettier.io/docs/en/ignore.html#ignoring-files-prettierignore) already does this by default:

> Prettier will also follow rules specified in the ".gitignore" file if it exists in the same directory from which it is run.

Related issues:

- https://github.com/eslint/eslint/issues/9329

Responding to this comment https://github.com/eslint/eslint/issues/9329#issuecomment-331484252:

I think it's a good idea to ignore all of the `.gitignore`, etc files by default. I think that can help a lot, clean up files, and remove the maintenance needs as I explained above. Not just `.gitignore` Let's do it for other things like SVN too!

@platinumazure

**Comment by nzakas** [comment]
Thanks for the suggestion. We've already considered this a few times and have decided not to read from `.gitignore` by default. It would dramatically complicate the already complicated config system for very little gain. Instead, you read your `.gitignore` file and inject that into your `eslint.config.js` file.
